htdocs/product/class/api_products.class.php

```diff
@@ -249,48 +249,55 @@ public function put($id, $request_data = null)
-                    }
-                    if ($this->product->price_min != $oldproduct->price_min) { $pricemodified = true;
-                    }
-                }
-            }
-
-            if ($pricemodified) {
-                $newvat = $this->product->tva_tx;
-                $newnpr = $this->product->tva_npr;
-                $newvatsrccode = $this->product->default_vat_code;
-
-                $newprice = $this->product->price;
-                $newpricemin = $this->product->price_min;
-                if ($this->product->price_base_type == 'TTC') {
-                    $newprice = $this->product->price_ttc;
-                    $newpricemin = $this->product->price_min_ttc;
-                }
-
-                $result = $this->product->updatePrice($newprice, $this->product->price_base_type, DolibarrApiAccess::$user, $newvat, $newpricemin, 0, $newnpr, 0, 0, array(), $newvatsrccode);
+        if ($result > 0) {
+        	// If price mode is 1 price per product
+			if (! empty($conf->global->PRODUCT_PRICE_UNIQ)) {
+				// We update price only if it was changed
+				$pricemodified = false;
+				if ($this->product->price_base_type != $oldproduct->price_base_type) { $pricemodified = true;
+				} else
+				{
+					if ($this->product->tva_tx != $oldproduct->tva_tx) { $pricemodified = true;
+					}
+					if ($this->product->tva_npr != $oldproduct->tva_npr) { $pricemodified = true;
+					}
+					if ($this->product->default_vat_code != $oldproduct->default_vat_code) { $pricemodified = true;
+					}
+
+					if ($this->product->price_base_type == 'TTC') {
+						if ($this->product->price_ttc != $oldproduct->price_ttc) { $pricemodified = true;
+						}
+						if ($this->product->price_min_ttc != $oldproduct->price_min_ttc) { $pricemodified = true;
+						}
+					}
+					else
+					{
+						if ($this->product->price != $oldproduct->price) { $pricemodified = true;
+						}
+						if ($this->product->price_min != $oldproduct->price_min) { $pricemodified = true;
+						}
+					}
+				}
+
+				if ($pricemodified) {
+					$newvat = $this->product->tva_tx;
+					$newnpr = $this->product->tva_npr;
+					$newvatsrccode = $this->product->default_vat_code;
+
+					$newprice = $this->product->price;
+					$newpricemin = $this->product->price_min;
+					if ($this->product->price_base_type == 'TTC') {
+						$newprice = $this->product->price_ttc;
+						$newpricemin = $this->product->price_min_ttc;
+					}
+
+					$result = $this->product->updatePrice($newprice, $this->product->price_base_type, DolibarrApiAccess::$user, $newvat, $newpricemin, 0, $newnpr, 0, 0, array(), $newvatsrccode);
+				}
+			}
+			
+        	// Update product type
+            if ($this->product->type != $oldproduct->type && is_numeric($this->product->type) && in_array((int) $this->product->type, [0,1])) {
+                $this->product->setValueFrom('fk_product_type', $this->product->type, '', null, 'text', '', DolibarrApiAccess::$user, 'PRODUCT_MODIFY');
```

Comment:
This will call the trigger PRODUCT_MODIFY a second time (already called by the update).

I think we can enhance the method "update" of product to add a parameter "updatetype" (not done by default and not done if property type is not provided in API) ?

Reply:
Effectivement, ce serait beaucoup mieux. J'ai copié bêtement ce qui existait déjà, je ne m'étais pas rendu compte que ça relançait la méthode update.

Reply:
si tu peux supprimer ce pull request. J'y comprends que dalle à github. Je vais recommencer.